hail/python/hailtop/batch_client/aioclient.py

```diff
@@ -443,55 +454,85 @@ def create_job(self, image, command, env=None, mount_docker_socket=False,
             f'/api/v1alpha/batches/{batch_id}/jobs/create',
             data=aiohttp.BytesPayload(
                 b, content_type='application/json', encoding='utf-8'))
+        self._byte_job_spec_bunches[bunch_index] = None
 
-    async def submit(self):
-        if self._submitted:
-            raise ValueError("cannot submit an already submitted batch")
-        self._submitted = True
+    async def _create(self):
+        assert self._batch is None
+        if self._batch_token is None:
+            self._batch_token = secrets.token_urlsafe(32)
 
-        batch_spec = {'billing_project': self._client.billing_project, 'n_jobs': len(self._job_specs)}
+        batch_spec = {'billing_project': self._client.billing_project,
+                      'n_jobs': len(self._job_specs),
+                      'token': self._batch_token}
         if self.attributes:
             batch_spec['attributes'] = self.attributes
         if self.callback:
             batch_spec['callback'] = self.callback
 
-        b_resp = await self._client._post('/api/v1alpha/batches/create', json=batch_spec)
-        b = await b_resp.json()
-        log.info(f'created batch {b["id"]}')
-        batch = Batch(self._client, b['id'], self.attributes)
-
-        byte_job_specs = [json.dumps(job_spec).encode('utf-8') for job_spec in self._job_specs]
-
-        groups = []
-        group = []
-        group_size = 0
-        for spec in byte_job_specs:
-            n = len(spec)
-            if group_size + n < 8000000 and len(group) < 1000:
-                group.append(spec)
-                group_size += n
-            else:
-                groups.append(group)
-                group = [spec]
-                group_size = n
-        if group:
-            groups.append(group)
+        batch_json = await (await self._client._post('/api/v1alpha/batches/create',
+                                                     json=batch_spec)).json()
+        return Batch(self._client, batch_json['id'], self.attributes)
 
-        await bounded_gather(*[functools.partial(self._submit_jobs, batch.id, group)
-                               for group in groups],
-                             parallelism=50)
+    async def submit(self,
+                     max_bunch_bytesize=8 * 1024 * 1024,
+                     max_bunch_size=8 * 1024,
+                     fail_on_first_bunch_failure=True):
```

Comment:
I thought you were going to remove this since only `True` makes sense here.  I think you can remove MultipleException in that case, too.

Reply:
ok